### PR TITLE
Add module docstrings

### PR DESCRIPTION
**Pull request by apavlen**
# Pull Request

## Title

Add module docstrings for streamlit files, recommenders and simulator 

---

## Description

Fix docstring and pylint related checks for streamlit files, recommenders and simulator 

## Type of Change

- 🔄 Refactor  

---

## Testing

Unit tests 
---

## Additional Notes (optional)

I'm going through modules adding docs, but all changed files also require minor fixes by pylint. No logic is added, mostly check or safer refactoring

### CODE REVIEW

**Comment by bpkroth** [comment]
Curious, what version of python do you have? Wonder if it's why pylint is flagging different things now.

File: examples/streamlit/web_demo.py [hunk 1]
```diff
@@ -6,7 +6,22 @@
 # --------------------------------------------------------------------------
 #
 
-# pylint: disable=no-member # FIXME
```
Comment (bpkroth) [inline examples/streamlit/web_demo.py:9]: Did you mean to remove this?

Comment (bpkroth) [reply examples/streamlit/web_demo.py:9]: Might just need to be moved below the docstring for the module

**Comment by apavlen** [comment]
> Curious, what version of python do you have? Wonder if it's why pylint is flagging different things now.

Python 3.11.9

File: examples/streamlit/web_demo.py (hunk 1 above)
Comment (apavlen) [reply examples/streamlit/web_demo.py:9]: Yeah..  I made some changes and wanted to remove some "ignores"

**Comment by bpkroth** [comment]
> > Curious, what version of python do you have? Wonder if it's why pylint is flagging different things now.
> 
> Python 3.11.9

Ah, yeah, I think we're only using 3.10 in the pipelines so that's all I was doing locally.  We might want to expand the pipelines to add that too.

File: src/vasim/simulator/InMemorySimulator.py [hunk 2]
```diff
@@ -130,14 +166,11 @@ def _initialize_output_file(data_dir):
...
-            f = open(out_file, "a", encoding="utf-8")
-            f.write("LATEST_TIME,CURR_LIMIT,NEW_LIMIT\n")
-            f.flush()
-        else:
-            f = open(out_file, "a", encoding="utf-8")
-
-        return f
+            with open(out_file, "a", encoding="utf-8") as f:
+                f.write("LATEST_TIME,CURR_LIMIT,NEW_LIMIT\n")
+        return open(out_file, "a", encoding="utf-8")
```
Comment (bpkroth) [inline src/vasim/simulator/InMemorySimulator.py:173]: I'm not certain this really solves the fact that we return an open file handle and never close it.

One way you could work around this though is as follows (basically a ContextManager pattern):

```suggestion
        with open(out_file, "a", encoding="utf-8") as f:
            yield f
```

File: src/vasim/simulator/InMemorySimulator.py [hunk 3]
```diff
@@ -130,14 +166,11 @@ def _initialize_output_file(data_dir):
...
+        # Use 'with' statement to open the file
         if not out_file.exists():
-            f = open(out_file, "a", encoding="utf-8")
-            f.write("LATEST_TIME,CURR_LIMIT,NEW_LIMIT\n")
-            f.flush()
-        else:
-            f = open(out_file, "a", encoding="utf-8")
-
-        return f
+            with open(out_file, "a", encoding="utf-8") as f:
```
Comment (bpkroth) [inline src/vasim/simulator/InMemorySimulator.py:171]:
```suggestion
            with open(out_file, "w", encoding="utf-8") as f:
```

File: .vscode/launch.json [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "Python Debugger: Current File",
+      "type": "debugpy",
+      "request": "launch",
+      "program": "${file}",
+      "console": "integratedTerminal"
```
Comment (bpkroth) [inline .vscode/launch.json:12]: Did this work?  If not, maybe drop this change?  Or put it in a separate PR at least
